### PR TITLE
bug：metrics插件趋势错误流水线构建页面的错误码要按错误类型查询 

### PR DESCRIPTION
**Pull request by yjieliang**

### CODE REVIEW

File: src/backend/ci/core/metrics/api-metrics/src/main/kotlin/com/tencent/devops/metrics/api/UserErrorCodeInfoResource.kt [hunk 1]
```diff
@@ -61,9 +62,12 @@ interface UserErrorCodeInfoResource {
         @ApiParam("userId", required = true)
         @HeaderParam(AUTH_HEADER_USER_ID)
         userId: String,
-        @ApiParam("错误类型", required = false)
+        @ApiParam("错误类型,使用逗号进行分隔", required = true)
```
Comment (carlyin0801) [inline src/backend/ci/core/metrics/api-metrics/src/main/kotlin/com/tencent/devops/metrics/api/UserErrorCodeInfoResource.kt:65]: 注释错了

File: src/backend/ci/core/metrics/biz-metrics/src/main/kotlin/com/tencent/devops/metrics/service/impl/ErrorCodeInfoServiceImpl.kt [hunk 2]
```diff
@@ -67,4 +82,60 @@ class ErrorCodeInfoServiceImpl @Autowired constructor(
             )
         )
     }
+
+    override fun syncAtomErrorCodeRel(userId: String): Boolean {
+        var offset = 0
+        val limit = 10
+        Executors.newFixedThreadPool(5).submit {
```
Comment (carlyin0801) [inline src/backend/ci/core/metrics/biz-metrics/src/main/kotlin/com/tencent/devops/metrics/service/impl/ErrorCodeInfoServiceImpl.kt:89]: 执行一个异步任务，要新建一个5个线程的线程池干什么

File: src/backend/ci/core/metrics/biz-metrics/src/main/kotlin/com/tencent/devops/metrics/service/impl/ErrorCodeInfoServiceImpl.kt [hunk 3]
```diff
@@ -67,4 +82,60 @@ class ErrorCodeInfoServiceImpl @Autowired constructor(
...
+            do {
+                val atomCodes = atomFailInfoDao.limitAtomCodes(dslContext, offset, limit)
```
Comment (carlyin0801) [inline src/backend/ci/core/metrics/biz-metrics/src/main/kotlin/com/tencent/devops/metrics/service/impl/ErrorCodeInfoServiceImpl.kt:91]: 不能直接按插件分页查询，现在是做了分库分表，直接按插件分页查询效在大数据量的情况下率很低，应该按分片键项目标识，一个个项目处理避免查所有的库和所有的表。

**Comment by tencentblueking-adm** [comment]
[CLA assistant check](https://bk-cla.bktencent.com:443/TencentBlueKing/bk-ci?pullRequest=8212) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://bk-cla.bktencent.com:443/TencentBlueKing/bk-ci?pullRequest=8212) before we can accept your contribution.<br/>**1** out of **2** committers have signed the CLA.<br/><br/>:white_check_mark: yjieliang<br/>:x: carlyin0801<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://bk-cla.bktencent.com:443/check/TencentBlueKing/bk-ci?pullRequest=8212) it.</sub>

**Review by carlyin0801 — APPROVED** [review]
ok

File: support-files/sql/2010_ci_metrics-update_v1.10_mysql.sql [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+    IF NOT EXISTS(SELECT 1
+                      FROM information_schema.COLUMNS
+                      WHERE TABLE_SCHEMA = db
+                        AND TABLE_NAME = 'T_ERROR_CODE_INFO'
+                        AND COLUMN_NAME = 'ATOM_CODE') THEN
+    ALTER TABLE `T_ERROR_CODE_INFO`
+        ADD COLUMN `ATOM_CODE` varchar(64) DEFAULT NULL COMMENT '关联插件代码';
+    END IF;
+
+    COMMIT;
```
Comment (carlyin0801) [inline support-files/sql/2010_ci_metrics-update_v1.10_mysql.sql:50]: 索引改了也要补充变动sql

**Review by irwinsun — APPROVED** [review]

**Pull request merged**